servicecontrol/contracts.md

```diff
@@ -42,10 +42,16 @@ To subscribe to the `MessageFailed` event:
 
  * Create an [NServiceBus endpoint](/nservicebus/hosting/nservicebus-host/).
  * Install the [ServiceControl.Contracts NuGet package](https://www.nuget.org/packages/ServiceControl.Contracts/).
- * Add the message mapping in the `UnicastBusConfig` section of the endpoint's app.config so that this endpoint will subscribe to the events from ServiceControl as shown:
+ * For transports using the [message-driven publish-subscribe](/nservicebus/messaging/publish-subscribe/) mechanism, either
+   * For NServiceBus Version 6 and higher, register the publisher by using the [routing config] code API(/nservicebus/messaging/routing.md#event-routing-message-driven).
+
+snippet: ServiceControlPublisherConfig
+
+   * or for NServiceBus Version 5 and below, add the message mapping in the `UnicastBusConfig` section of the endpoint's app.config so that this endpoint will subscribe to the events from ServiceControl as shown:
 
 snippet: ServiceControlEventsXmlConfig
 
+ * Transports that natively support publish and subscribe, see [multicast-enabled transports](/transports/#types-of-transports-multicast-enabled-transports), do not require to map the contracts.
```

Comment:
"do not require to" isn't quite the right way to phrase that.

Something like "do not require any additional configuration" flows better.